specs/core/0_beacon-chain.md

```diff
@@ -1726,17 +1727,24 @@ def process_attestation(state: BeaconState, attestation: Attestation) -> None:
 ##### Deposits
 
 ```python
-def process_deposit(state: BeaconState, deposit: Deposit) -> None:
+def process_deposit(state: BeaconState,
+                    deposit: Deposit,
+                    deposit_index: Optional[uint64]=None,
+                    deposit_root: Optional[Hash]=None) -> None:
```

Comment:
No need to add `deposit_index` and `deposit_root` :) It suffices to properly initialise `state.eth1_data.deposit_root` corresponding to the empty sparse Merkle tree after `state = BeaconState()` in `is_genesis_trigger`.

Reply:
A mathematically clean definition would be `state.eth1_data.deposit_root = hash_tree_root(Vector[Hash, 2** DEPOSIT_CONTRACT_TREE_DEPTH])` but that would stall any implementation that doesn't implement the sparse Merkle tree optimisations.

Reply:
1. Right! But that means we have to initialize and update `Eth1Data` for *each* genesis deposit in both `is_genesis_trigger` and `get_genesis_beacon_state`... but I agreed that making `process_deposit` clean is a win for the long term!
2. Also note that there will be so many different ways to optimize `is_genesis_trigger` in client side. 😄